srtcore/logging.h

```diff
@@ -51,7 +51,12 @@ written by
 // LOGC uses an iostream-like syntax, using the special 'log' symbol.
 // This symbol isn't visible outside the log macro parameters.
 // Usage: LOGC(mglog.Debug, log << param1 << param2 << param3);
-#define LOGC(logdes, args) if (logdes.CheckEnabled()) { srt_logging::LogDispatcher::Proxy log(logdes); log.setloc(__FILE__, __LINE__, __FUNCTION__); args; }
+#define LOGC(logdes, args) if (logdes.CheckEnabled()) \
```

Comment:
This change improves the compiler error report for a case when someone forgot to pass the `log` marker in the call to `LOGC` and `HLOGC` macros. This enforces the interpretation of `args` as convertible to the type of `log` so that when someone mistakenly writes `LOGC(mglog.Debug, "ACK received")` the compiler will complain about inability to convert the string constant into the type of `log`.